test/loader.spec.js

```diff
@@ -1,219 +1,196 @@
-const { describe, it, afterEach, expect } = require("@jest/globals");
-const fsPromises = require("fs").promises;
-const path = require("path");
-const webpack = require("webpack");
-const xmldom = require("xmldom");
-
-const outputDir = path.resolve(__dirname, "./output");
-const bundleFileName = "bundle.js";
-const svgAsSymbolLoader = path.resolve(__dirname, "../");
-
-function getBundleFile() {
-  return path.join(outputDir, bundleFileName);
-}
+const { describe, it, afterEach, expect, afterAll } = require('@jest/globals')
+const { rejects } = require('assert')
+// const util = require('util');
+// const fs = require('fs');
+// const readdir = util.promisify(fs.readdir);
+// const unlink = util.promisify(fs.unlink);
+const path = require('path')
+const webpack = require('webpack')
+const xmldom = require('xmldom')
+
+const outputDir = path.resolve(__dirname, './output')
+
+const svgAsSymbolLoader = path.resolve(__dirname, '../')
 
 function getBaseWebpackConfig() {
   return {
-    context: path.resolve(__dirname, "../"),
-    mode: "development",
+    context: path.resolve(__dirname, '../'),
+    mode: 'development',
     devtool: false,
+    cache: false,
     output: {
       path: outputDir,
-      filename: bundleFileName,
-      publicPath: "",
-      libraryTarget: "commonjs2",
+      clean: true,
+      filename: 'bundle.[fullhash].js',
+      publicPath: '',
+      libraryTarget: 'commonjs2'
     },
     module: {
       rules: [
         {
           test: /\.svg/,
           exclude: /node_modules/,
           loader: svgAsSymbolLoader,
-          options: {},
-        },
-      ],
-    },
-  };
+          options: {}
+        }
+      ]
+    }
+  }
 }
 
 async function runWebpack(config) {
   return new Promise((resolve, reject) => {
-    webpack(config, function (err, data) {
+    webpack(config, function (err, stats) {
       if (err) {
-        reject(err);
+        reject(err)
       } else {
-        resolve(data);
+        resolve(stats.toJson('minimal').assetsByChunkName.main[0])
       }
-    });
-  });
+    })
+  })
 }
 
-async function evaluateGeneratedBundle() {
-  const buffer = await fsPromises.readFile(getBundleFile());
-  const content = buffer.toString();
-  return eval(content);
+function evaluateGeneratedBundle(assetName) {
+  function getBundleFile(assetName) {
+    return path.join(outputDir, assetName)
+  }
+  // require is now mandatory with webpack >5.22.0
+  return eval(`require("${getBundleFile(assetName)}")`)
 }
 
-describe("svg-as-symbol-loader", function () {
-  // Clean generated cache files before each test so that we can call each test with an empty state.
-  afterEach(() => fsPromises.unlink(getBundleFile()));
```

Comment:
Does removing this mean that after each test run some files will be generated and left on a filesystem?

If so, can you please change webpack config to store them inside some folder and remove that folder in `afterAll` ?

Reply:
The bundle file generated is different now for any test. This solves the trouble with module cache. 
The new `clean: true` will clean the output directory, and the asset name will have the hash in the name so it won't be possible to recycle the same bundle.